solr/core/src/java/org/apache/solr/search/join/ChildFieldValueSourceParser.java

```diff
@@ -133,6 +134,7 @@ public SortField getSortField(boolean reverse) {
         @Override
         public FieldComparator<?> getComparator(int numHits, int sortPos) {
           final FieldComparator<?> comparator = super.getComparator(numHits, sortPos);
+          // SOLR-16172 Returning the default comparator fixes the crash, why was this needed?
           return type == Type.STRING
               ? new BytesToStringComparator((FieldComparator<BytesRef>) comparator)
               : comparator;
```

Comment:
@mkhludnev Why was this custom BytesToStringComparator needed? Removing it fixes this bug...